### PR TITLE
Convert copy-constraint padding partitions into an enum

### PR DESCRIPTION
**Pull request by liliatangxy**
There are currently only two modes of cc padding partitions used, so we can convert it into an enum to reduce memory

### CODE REVIEW

File: src/basic_block/copy_constraint.rs [hunk 1]
```diff
@@ -103,16 +104,56 @@ pub fn zero_padding_partition(permutation: &ArrayD<Option<IxDyn>>) -> HashMap<Fr
...
+  Zero,
+  Max(Fr),
+}
+
+// Gets padding partition based on the enum value
+fn get_padding_partition(permutation: &ArrayD<Option<IxDyn>>, pad_type: &PaddingEnum) -> HashMap<Fr, Vec<IxDyn>> {
+  match pad_type {
+    PaddingEnum::Zero => zero_padding_partition(permutation),
+    PaddingEnum::Max(val) => max_padding_partitions(permutation, *val),
+  }
```
Comment (Vimalakirti) [inline src/basic_block/copy_constraint.rs:156]: Is it deterministic for setup, run, prove, and verify?

Comment (liliatangxy) [reply src/basic_block/copy_constraint.rs:156]: For setup and run, the hashmap order is not relevant. For prove and verify, the fix you made previously with sorting the keys should still work.

Comment (Vimalakirti) [reply src/basic_block/copy_constraint.rs:156]: How about the order of `Vec<IxDyn>`?

Comment (liliatangxy) [reply src/basic_block/copy_constraint.rs:156]: I think the ndarray iterators are deterministic. They satisfy row-major order: https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#indexing-and-dimension

**Review by Vimalakirti — APPROVED** [review]

**Review by ddkang — APPROVED** [review]

**Pull request merged**